### PR TITLE
Add `hibou` models to registry

### LINKED ISSUE #608: Onboard `hibou` models from HistAI
We should add the `hibou-B` and `hibou-L` models from https://github.com/HistAI/hibou to our model registry & leaderboard.

### PR DESCRIPTION
**Pull request by nkaenzig**
Closes #608

### CODE REVIEW

File: docs/user-guide/advanced/replicate_evaluations.md [hunk 1]
```diff
@@ -140,6 +140,30 @@ IN_FEATURES=1024 \
 eva predict_fit --config configs/vision/pathology/offline/<task>.yaml
 ```
 
+### hibou-B (hist.ai) - DINOv2 ViT-B14 (1M Slides) [[7]](#references)
+To evaluate [hist.ai's](https://www.hist.ai/) FM with DINOv2 ViT-B14 backbone, pretrained on
+a proprietary dataset of one million slides, run:
```
Comment (roman807) [inline docs/user-guide/advanced/replicate_evaluations.md:145]: add link to model card, e.g.: `.... slides, available for download on [huggingface](https://huggingface.co/histai/hibou-b), run:`

File: docs/user-guide/advanced/replicate_evaluations.md [hunk 2]
```diff
@@ -140,6 +140,30 @@ IN_FEATURES=1024 \
...
+MODEL_NAME=pathology/histai_hibou_b \
+NORMALIZE_MEAN=[0.7068,0.5755,0.722] \
+NORMALIZE_STD=[0.195,0.2316,0.1816] \
+IN_FEATURES=768 \
+eva predict_fit --config configs/vision/pathology/offline/<task>.yaml
+```
+
+### hibou-L (hist.ai) - DINOv2 ViT-L14 (1M Slides) [[7]](#references)
+To evaluate [hist.ai's](https://www.hist.ai/) FM with DINOv2 ViT-L14 backbone, pretrained on
+a proprietary dataset of one million slides, run:
```
Comment (roman807) [inline docs/user-guide/advanced/replicate_evaluations.md:157]: mention that histai needs to be contacted for download and link to https://www.hist.ai/#models

File: src/eva/vision/models/networks/backbones/_utils.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    model_name: str,
+    out_indices: int | Tuple[int, ...] | None,
+    model_kwargs: Dict[str, Any] | None = None,
+    transform_args: Dict[str, Any] | None = None,
+) -> nn.Module:
+    """Helper function to load HuggingFace models.
+
+    Args:
+        model_name: The model name to load.
+        out_indices: Weather and which multi-level patch embeddings to return.
```
Comment (roman807) [inline src/eva/vision/models/networks/backbones/_utils.py:21]: `Whether`

**Review by roman807 — APPROVED** [review]

File: docs/user-guide/advanced/replicate_evaluations.md (hunk 2 above)
Comment (nkaenzig) [reply docs/user-guide/advanced/replicate_evaluations.md:157]: I think the website is not up to date, the model is available on huggingface, same as `hubou-B`
https://huggingface.co/histai/hibou-L

File: src/eva/vision/models/networks/backbones/_utils.py (hunk 3 above)
Comment (nkaenzig) [reply src/eva/vision/models/networks/backbones/_utils.py]: done

File: docs/user-guide/advanced/replicate_evaluations.md (hunk 1 above)
Comment (nkaenzig) [reply docs/user-guide/advanced/replicate_evaluations.md]: done
